File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        n = len(heights)
        i=0 
        j=n-1

        mx = -1

        while(i<j):
            mx = max(min(heights[i],heights[j])*(j-i),mx)
            if(heights[i]<heights[j]):
                i+=1;
            else:
                j-=1;
        return mx
        

                       

File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:

    #define INF 10000007
    int jump(vector<int>& nums) {
        int n = nums.size();

        vector<int> prefmx(n,0);
        prefmx[0] = nums[0];
        for(int i =1 ;i<n; i++) {
            prefmx[i] = max(prefmx[i-1], i+nums[i]);
        }

        int mx = 0;
        int ct = 0;
        while(mx < n-1) {
            ct++;   
            mx = prefmx[mx];
        }
        return ct;
    }
};
